src/test/java/simplifier/controllers/LinkControllerIT.java

```diff
@@ -137,38 +162,44 @@ public void conflictIfShortenedNotUnique() throws Exception {
 
     @Test
     public void getLinksByTag() throws Exception{
-        User user = new User();
-        user.setUsername("author");
-        userRepository.save(user);
+        Link link = initLink();
 
-        LinkCreationDto link = mapper.readValue(new File("src/test/resources/validLink.JSON"),
-                LinkCreationDto.class);
-        ResponseEntity<LinkGetterDto>  createdLink = restTemplate.postForEntity("/links", link,
-                LinkGetterDto.class);
         LinkGetterDto[] response = restTemplate.getForObject("/links/tags/1",
                 LinkGetterDto[].class);
+
         assertThat(response.length, is(1));
         LinkGetterDto responseLink = response[0];
 
-        assertThat(responseLink, is(createdLink.getBody()));
+        assertThat(responseLink.getOriginalLink(), is(link.getOriginalLink()));
+        assertThat(responseLink.getShortenedLink(), is(link.getShortenedLink()));
+        assertThat(responseLink.getDescription(), is(link.getDescription()));
+        assertThat(responseLink.getAuthor(), is(link.getAuthor().getUsername()));
+        assertThat(responseLink.getClicks(), is(link.getClicks()));
+        List<String> tags = link.getTags().stream()
+                .map(tag -> tag.getName())
+                .collect(Collectors.toList());
+        assertThat(responseLink.getTags(), is(tags));
 
     }
 
     @Test
     public void getLinksByUser() throws Exception{
-        User user = new User();
-        user.setUsername("author");
-        userRepository.save(user);
+        Link link = initLink();
 
-        LinkCreationDto link = mapper.readValue(new File("src/test/resources/validLink.JSON"),
-                LinkCreationDto.class);
-        ResponseEntity<LinkGetterDto>  createdLink = restTemplate.postForEntity("/links", link,
-                LinkGetterDto.class);
         LinkGetterDto[] response = restTemplate.getForObject("/links/users/author",
                 LinkGetterDto[].class);
         assertThat(response.length, is(1));
         LinkGetterDto responseLink = response[0];
-        assertThat(responseLink, is(createdLink.getBody()));
+
+        assertThat(responseLink.getOriginalLink(), is(link.getOriginalLink()));
+        assertThat(responseLink.getShortenedLink(), is(link.getShortenedLink()));
+        assertThat(responseLink.getDescription(), is(link.getDescription()));
+        assertThat(responseLink.getAuthor(), is(link.getAuthor().getUsername()));
+        assertThat(responseLink.getClicks(), is(link.getClicks()));
+        List<String> tags = link.getTags().stream()
+                .map(tag -> tag.getName())
+                .collect(Collectors.toList());
+        assertThat(responseLink.getTags(), is(tags));
```

Comment:
move asserts about verifying link is the same to private method or create hamcrest matcher